apiserver/facades/client/application/application_get.go

```diff
@@ -73,9 +73,17 @@ func (api *APIBase) getConfig(
 	appConfigInfo := describeAppConfig(appConfig, providerSchema, providerDefaults)
 	var cons constraints.Value
 	if app.IsPrincipal() {
-		cons, err = app.Constraints()
-		if err != nil {
-			return params.ApplicationGetResults{}, err
+		appID, err := api.applicationService.GetApplicationIDByName(ctx, args.ApplicationName)
+		if errors.Is(err, applicationerrors.ApplicationNotFound) {
+			return params.ApplicationGetResults{}, errors.NotFoundf("application %s", args.ApplicationName)
+		} else if err != nil {
+			return params.ApplicationGetResults{}, errors.Trace(err)
+		}
+		cons, err = api.applicationService.GetApplicationConstraints(ctx, appID)
```

Comment:
This should have been just the name, not the application ID, similar to charm, use the natural key.

Reply:
` GetApplicationConfig()`, `GetApplicationTrustSetting()`, ` UnsetApplicationConfigKeys()`, `SetApplicationConfig()` all these methods use app ID as ident. There is some homogenization work on the app domain but it should not be part of this patch.